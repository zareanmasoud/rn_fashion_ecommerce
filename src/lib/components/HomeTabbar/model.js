// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
