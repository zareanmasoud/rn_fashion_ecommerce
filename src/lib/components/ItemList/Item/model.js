// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
