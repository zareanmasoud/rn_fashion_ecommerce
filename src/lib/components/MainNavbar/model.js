// @flow

export type Button = {
  action: Function,
  src: string,
};
