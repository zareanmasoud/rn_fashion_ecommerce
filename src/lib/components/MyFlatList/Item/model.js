// @flow

export type Data = {
  id: string,
  photo: string,
};
