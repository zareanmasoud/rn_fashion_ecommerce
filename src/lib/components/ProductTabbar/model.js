// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
