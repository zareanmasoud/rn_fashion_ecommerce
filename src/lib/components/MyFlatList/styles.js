import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  contentContainer: {},
});

export default styles;
