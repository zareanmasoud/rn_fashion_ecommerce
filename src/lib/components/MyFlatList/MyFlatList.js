import React from 'react';
import {FlatList} from 'react-native';
import styles from './styles';
import type {Data} from './Item/types';

type Props = {
  data: Data[],
  itemComponent: React.Node,
};

export default function MyFlatList({data, itemComponent}: Props) {
  const Item = itemComponent;

  return (
    <FlatList
      contentContainerStyle={styles.contentContainer}
      data={data}
      keyExtractor={(item, index) => index.toString()}
      // scrollEnabled={false}
      renderItem={({item, index}) => <Item data={item} index={index} />}
    />
  );
}
