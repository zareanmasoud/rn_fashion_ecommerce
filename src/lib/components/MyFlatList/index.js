import MyFlatList from './MyFlatList';

export {MyFlatList as default};
