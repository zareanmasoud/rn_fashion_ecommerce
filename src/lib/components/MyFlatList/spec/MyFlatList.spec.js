import React from 'react';
import {shallow} from 'enzyme';
import {View} from 'react-native';
import MyFlatList from '../MyFlatList';
import model from './model';

describe('MyFlatList Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(
        <MyFlatList data={model.products} itemComponent={<View />} />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
