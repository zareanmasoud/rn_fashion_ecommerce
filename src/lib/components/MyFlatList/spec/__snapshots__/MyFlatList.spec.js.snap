// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFlatList Component Rendering should match to snapshot 1`] = `
<FlatList
  contentContainerStyle={Object {}}
  data={
    Array [
      Object {
        "description": "Description",
        "discount_code": "X9E2AD",
        "id": "1",
        "photo": Object {
          "process": [Function],
        },
        "title": "Title",
      },
      Object {
        "description": "Description",
        "discount_code": "X9E2AD",
        "id": "2",
        "photo": Object {
          "process": [Function],
        },
        "title": "Title",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
